src/components/Card.js

```diff
@@ -13,36 +25,86 @@ export default class Card {
+      "card__like-button-active"
+    );
+    this._isLiked = isLiked;
+    this._handleLikeCard(this);
   };
 
+  updateLikes(likes) {
+    // console.log("update like", likes.length);
+    this._cardCountEl.textContent = likes.length;
+    // console.log(likes, this._cardData.likes);
+    this._cardData.likes = likes;
+    this._renderLikes();
+  }
+
+  isLiked() {
+    // console.log(this._cardData.likes);
+    for (let i = 0; i < this._cardData.likes.length; i++) {
+      // console.log(this._cardData.likes[i]._id);
+      if (this._userId === this._cardData.likes[i]._id) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  _renderLikes() {
+    if (this.isLiked()) {
+      this._likeButton.classList.add("card__like-button-active");
+    } else {
+      this._likeButton.classList.remove("card__like-button-active");
+    }
+  }
+
   _openPreview = () => {
     this._handleCardClick(this._cardData);
   };
 
-  _setEventListeners(cardElement, cardData) {
-    this._likeButton.addEventListener("click", this._toggleLike);
-
-    const deleteButton = cardElement.querySelector(".card__delete");
-    deleteButton.addEventListener("click", this._clickDeleteButton);
+  _clickDeleteButton = () => {
+    console.log("_clickDeleteButton called ", this._handleCardDelete);
+    console.log(this._cardElement);
+    this._handleCardDelete(this._cardData, this._cardElement);
+    // e.target.closest(".card").remove();
+  };
 
+  _setEventListeners() {
+    this._likeButton.addEventListener("click", this._toggleLike);
+    this._deleteButton.addEventListener("click", this._clickDeleteButton);
     this._cardImageEl.addEventListener("click", this._openPreview);
   }
 
   createCardElement() {
-    const cardElement = this._getTemplate().cloneNode(true);
-    const cardTitleEl = cardElement.querySelector(".card__title");
+    this._cardElement = this._getTemplate().cloneNode(true);
+    const cardTitleEl = this._cardElement.querySelector(".card__title");
     cardTitleEl.textContent = this._cardData.name;
-    this._cardImageEl = cardElement.querySelector(".card__image");
+    this._cardImageEl = this._cardElement.querySelector(".card__image");
+    this._cardCountEl = this._cardElement.querySelector(".card__like-count");
     this._cardImageEl.src = this._cardData.link;
+    this._cardImageEl.id = this._cardData._id;
     this._cardImageEl.alt = this._cardData.name;
-    this._likeButton = cardElement.querySelector(".card__like-button");
-    this._setEventListeners(cardElement, this._cardData);
-    return cardElement;
-  }
-
-  _clickDeleteButton(e) {
-    e.target.closest(".card").remove();
+    this._cardCountEl.textContent = this._cardData.likes.length;
+    this._likeButton = this._cardElement.querySelector(".card__like-button");
+    this._deleteButton = this._cardElement.querySelector(".card__delete");
+    if (this._cardData.owner._id !== this._userId) {
+      this._deleteButton.style.display = "none";
```

Comment:
Please remove the button from the DOM, so a user won't be able to click on it even using the console